### PR TITLE
Add example for wrap.sh based HWASan

### PR DESCRIPTION
**Pull request by fmayer**

### CODE REVIEW

**Comment by fmayer** [comment]
@emrekultursay FYI. The problem here is that I didn't know how to change the API level for the HWASan build only.

**Comment by emrekultursay** [comment]
Try adding this to your `build.gradle` in the `android {...}` block:

```
    androidComponents {
        beforeVariants(selector().withName("hwasan")) {
            minSdk = 34
        }
    }
```

**Comment by fmayer** [comment]
> Try adding this to your `build.gradle` in the `android {...}` block:
> 
> ```
>     androidComponents {
>         beforeVariants(selector().withName("hwasan")) {
>             minSdk = 34
>         }
>     }
> ```

That works. PTAL.

**Comment by fmayer** [comment]
Tested by checking `__ANDROID_API__` in the cpp file.

**Review by emrekultursay — APPROVED** [review]
LGTM.

**Comment by fmayer** [comment]
@emrekultursay for some reason it says "At least 1 approving review is required by reviewers with write access", also the presubmit build is failing with unrelated lint errors. Do you know how to fix these? AFAICT you should have write access to this repo

**Comment by emrekultursay** [comment]
I am not an owner of this repo. I believe @DanAlbert or @enh-google can help.

**Comment by fmayer** [comment]
Rebased and presubmit checks are now passing, PTAL.

File: sanitizers/app/build.gradle [hunk 1]
```diff
@@ -25,8 +25,16 @@ android {
...
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
         }
+        // HWASan for devices starting from Android 14. Does no longer require a special system image.
+        // See https://developer.android.com/ndk/guides/hwasan.
         hwasan {
             initWith debug
+            debuggable true
+            packagingOptions {
+                jniLibs {
+                    useLegacyPackaging = true
```
Comment (DanAlbert) [inline sanitizers/app/build.gradle:38]: Is this actually needed? For wrap.sh maybe? Could you add a comment?

**Review by DanAlbert — CHANGES_REQUESTED** [review]
I expanded the comment attached to minSdk to explain that it's not always true. The other comment I had is one I'll have to leave to you though. Otherwise LGTM.

File: sanitizers/app/build.gradle (hunk 1 above)
Comment (fmayer) [reply sanitizers/app/build.gradle:36]: Yes, sadly.

**Review by DanAlbert — APPROVED** [review]

**Pull request merged**